JAICore/jaicore-ml/src/main/java/ai/libs/jaicore/ml/tsc/classifier/trees/LearnPatternSimilarityClassifier.java

```diff
@@ -189,19 +189,10 @@ public Integer predict(final List<double[]> multivInstance) throws PredictionExc
 	 */
 	@Override
 	public List<Integer> predict(final TimeSeriesDataset dataset) throws PredictionException {
-		if (!this.isTrained()) {
-			throw new PredictionException("Model has not been built before!");
-		}
-
-		if (dataset == null || dataset.isEmpty()) {
-			throw new IllegalArgumentException("Dataset to be predicted must not be null or empty!");
-		}
-
+		double[][] data = this.checkWhetherPredictionIsPossible(dataset);

```

Comment:
Such things should be avoided, imo. Firstly, the methods does two things, namely performing some check on the data and then returning it in a different form. Apart from that this is a bad coding style (imo), it is also hard to understand. The method name only suggests, that some check is performed. I would expect that it returns a boolean or throws an exception (incase the check fails), but not that it returns any data.